### ISSUE TITLE
✨ Change order pattern of all Confirm/Cancel buttons

### ISSUE THREAD

**Issue by Jerryescandon** [GCTC-NTGC/gc-digital-talent#5738]
## ✨ Feature

Design team has had lengthy discussions about the best approach at ordering buttons at the end of forms and modals, and we have landed on a left to right approach where the leftmost is the most likely/friendly/useful action and the rightmost is the least likely action.

https://www.figma.com/file/G7088DJsXxWY1l4kiGeD1x/Style-guide?type=design&node-id=166-3767&mode=design&t=1BRp0x4onnWBuXN8-0

## 🕵️ Details

This applies to the entire platform and dialogs.

## 📸 Screenshot

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/8978655/82eb4516-d8f2-43e5-ab59-cc447d4714aa)

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/8978655/8025c075-0ebf-4b40-9d2e-bca86333d534)

### Pair with Designer
- [ ] Go through app, making a decision about every case
- [ ] Figma updated at same time
### General Direction
- [ ] Update entire app, including dialogs
- [ ] The primary action of the task should use the solid button style
- [ ] Subsequent/secondary actions should use the inline button style
- [ ] The primary action should always come first, left aligned
- [ ] Subsequent/secondary actions should follow the primary action, left aligned
### Button and link colour
- [ ] Main task actions should use the secondary color
- [ ] Actions that cancel a task should use the warning color
- [ ] Actions that are destructive should use the error color
### Uncertainty
- [ ] If unsure how to proceed in a particular case, document it in a new ticket

**Comment by esizer** [comment]
This may be an issue in specific scenarios like when one of the options is destructive. For keyboard and AT users, it is best practice to lead with the least destructive action to prevent accidental destructive actions.

When leading with a non-destructive item, the user can skip over it and get to the desired action. If they accidently interact with a non-destructive action, there is no harm done. However, if they land on a destructive action initially and interact with it, they could accidently do something destructive. 
 
An example of this is our delete/logout alert dialogs. We put the "Cancel" button on the left so that a user does not accidently delete an item or logout. So even though logging out may be the most useful action when the alert appears, it is not when a user accidently invoked the alert. Even though logging back in isn't a monumental task to do if you accidently logout, it could add frustration on repeated accidental actions which can be avoided by leading with the least destructive action.


![Image](https://user-images.githubusercontent.com/4127998/225935173-41b83097-9422-4055-860b-d23c8cb9978b.png)

So when referring to forms to create/update the cancel button would be the more destructive act (losing form state). Where as, when deleting/logging out, the inverse is true. 

This is a great idea, we just need to be careful about the order instead of **_always_** putting a cancel button on the right.

**Comment by petertgiles** [comment]
@Jerryescandon do you agree with Eric's advice that for destructive actons like delete and logout that cancel should go to the left?

**Comment by Jerryescandon** [comment]
Yup, tbh I had heard points that argued the contrary, but those do not take into account A11y considerations. 

I am confident we can continue with Eric's suggestion. In the case of destructive actions, invert the priority of the buttons.

**Comment by mnigh** [comment]
@Jerryescandon I tried to take this issue on but I feel like even with @esizer's comments and the revised acceptance criteria, every single action set is subjective on the most to least destructive scale and the likely to least likely scale (and which scale takes precedent). Also, the positioning and colours of the action sets seems inconsistent, so I am not even sure where to start. I took screenshots as examples of these. Maybe the best approach would be to create an issue for each one with a clear direction for each?

## Screenshots
![Screen Shot 2023-11-09 at 12 04 40](https://github.com/GCTC-NTGC/gc-digital-talent/assets/3046459/b877efbb-bde2-4b8f-a735-b8ec1bfc8c94)
![Screen Shot 2023-11-09 at 12 09 10](https://github.com/GCTC-NTGC/gc-digital-talent/assets/3046459/3f3f405f-b95f-4acb-a70c-734d384e6b90)
![Screen Shot 2023-11-09 at 12 11 00](https://github.com/GCTC-NTGC/gc-digital-talent/assets/3046459/001aeca3-147a-48b8-b4b3-c10a1ce59b9c)
![Screen Shot 2023-11-09 at 12 14 41](https://github.com/GCTC-NTGC/gc-digital-talent/assets/3046459/6d6c3026-3553-4271-80a4-35c443d72455)
![Screen Shot 2023-11-09 at 12 15 39](https://github.com/GCTC-NTGC/gc-digital-talent/assets/3046459/89e942db-c1ca-4e25-a0e5-5bb8c2b89619)
![Screen Shot 2023-11-09 at 12 32 45](https://github.com/GCTC-NTGC/gc-digital-talent/assets/3046459/82fe1fd9-9666-4dd1-b9fe-dbd646d86a02)
![Screen Shot 2023-11-09 at 12 35 17](https://github.com/GCTC-NTGC/gc-digital-talent/assets/3046459/24f6d569-71aa-433a-a50f-047d008663b2)
![Screen Shot 2023-11-09 at 12 39 24](https://github.com/GCTC-NTGC/gc-digital-talent/assets/3046459/8dd052e1-c06f-4273-a9c6-f0606b0b78c4)
![Screen Shot 2023-11-09 at 13 19 44](https://github.com/GCTC-NTGC/gc-digital-talent/assets/3046459/d509072d-fff1-465f-a2b3-b689d15f632b)
![Screen Shot 2023-11-09 at 14 09 22](https://github.com/GCTC-NTGC/gc-digital-talent/assets/3046459/d714589e-5dad-4da1-9c33-b4b19343f062)
![Screen Shot 2023-11-09 at 14 38 08](https://github.com/GCTC-NTGC/gc-digital-talent/assets/3046459/3e843cea-1303-4de5-b104-c3d7676af0e9)
![Screen Shot 2023-11-09 at 11 51 12](https://github.com/GCTC-NTGC/gc-digital-talent/assets/3046459/e63196ce-653b-4b47-97cd-7567b8d16b4f)
![Screen Shot 2023-11-09 at 11 55 53](https://github.com/GCTC-NTGC/gc-digital-talent/assets/3046459/3764812f-8616-47db-a8e0-db3cd7603031)
![Screen Shot 2023-11-09 at 11 59 31](https://github.com/GCTC-NTGC/gc-digital-talent/assets/3046459/3fbf18f8-138d-4c48-bdd2-8ba55f86cecc)
![Screen Shot 2023-11-09 at 12 04 16](https://github.com/GCTC-NTGC/gc-digital-talent/assets/3046459/bb576448-20a8-41f8-aae5-66ca37d62ed8)

**Comment by gobyrne** [comment]
@substrae has just completed some guidance for this in the style guide:
https://www.figma.com/file/G7088DJsXxWY1l4kiGeD1x/Style-guide?type=design&node-id=166-3767&mode=design&t=ZhnvDeFo7ZmpKjlv-4

**Comment by gobyrne** [comment]
I updated the acceptance criteria. Should be unblocked from des.

**Comment by petertgiles** [comment]
Thanks!  I updated the issue description based on the style guide.
